### PR TITLE
Search all distinct drive id's rather than just default drive id for --get-file-link (Issue #1227)

### PR DESCRIPTION
**Pull request by abraunegg**
* Add new DB query to return distinct driveId's from database
* Use all distinct drive ID's to query if path is in-sync with OneDrive before attempting to retrieve the item file link from OneDrive

### CODE REVIEW

**Comment by abraunegg** [comment]
With a file that has been synced to OneDrive Business Shared Folder:

**Without PR:**
```
./onedrive --confdir '~/.config/onedrive-business/' --get-file-link './Top Folder/asdfasdfs.txt'
Configuration file successfully loaded
Configuring Global Azure AD Endpoints
File has not been synced with OneDrive: ./Top Folder/asdfasdfs.txt
```

**With PR:**
```
./onedrive --confdir '~/.config/onedrive-business/' --get-file-link './Top Folder/asdfasdfs.txt'
Configuration file successfully loaded
Configuring Global Azure AD Endpoints
https://<redacted>/Documents/Top%20Folder/asdfasdfs.txt
```

File: src/itemdb.d [hunk 1]
```diff
@@ -484,4 +484,18 @@ final class ItemDatabase
 		auto stmt = db.prepare("VACUUM;");
 		stmt.exec();
 	}
+	
+	// Select distinct driveId items from database
+	string[] selectDistinctDriveIds()
+	{
+		string[] driveIdArray;
+		auto stmt = db.prepare("SELECT DISTINCT driveId FROM item;");
+		auto res = stmt.exec();
```
Comment (norbusan) [inline src/itemdb.d:493]: Maybe we need to protect that against broken sql database and assertions, similar to another recent PR? (resync necessary)

Comment (abraunegg) [reply src/itemdb.d:493]: Not required here, as we are not building any complex path or complex additional SQL query based on the prior response.

`SELECT DISTINCT driveId FROM item;` will return zero rows or rows of distinct driveId's. If no rows are returned, an empty array is returned to the calling function, which has then no driveid's to iterate through, thus, path is not in sync in DB.

Comment (norbusan) [reply src/itemdb.d:493]: Ok, thank!

**Review by norbusan — APPROVED** [review]

**Pull request merged**